test/pokemon/stab.test.ts

```diff
@@ -110,9 +113,80 @@ describe("STAB", () => {
     expect(enemyPokemon.calcStabMultiplierForTakingDamage).toHaveReturnedWith(1.5);
   });
 
-  it.todo("should have a 1.5 STAB on pledge moves");
+  it("combined Pledge moves should have a 1.5 STAB regardless of the user's type", async () => {
+    game.override.battleType("double");
 
-  it.todo("should have a 2.0 STAB on pledge moves if tera type DOES NOT match default type");
+    await game.classicMode.startBattle([SpeciesId.CHARMANDER, SpeciesId.SQUIRTLE]);
 
-  it.todo("should have a 2.25 STAB on pledge moves if tera type does match default type");
+    const enemyPokemon = game.scene.getEnemyField();
```

Comment:
NIT: Use 
```suggestion
    const [, enemyPokemon] = game.scene.getEnemyField();
```
instead